sdk/identity/azure-identity/azure/identity/_credentials/shared_cache.py

```diff
@@ -60,7 +60,7 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
 
         # try each refresh token, returning the first access token acquired
         for refresh_token in self._get_refresh_tokens(account):
-            token = self._client.obtain_token_by_refresh_token(refresh_token, scopes)
+            token = self._client.obtain_token_by_refresh_token(scopes, refresh_token)
             return token
```

Comment:
Why we use obtain_token_by_refresh_token(refresh_token["secret"]) in _redeem_refresh_token but refresh_token (no ["secret"]) here?

Reply:
`_get_refresh_tokens` now returns the token as a string rather than MSAL's dict (see the diff for shared_token_cache.py).

Reply:
Can we rename it to something lie _get_refresh_token_secrect? Both use refresh_token but have different meanings/types is confusing.

Reply:
I agree it's confusing and I want to make it less so by having "refresh token" consistently refer to the token as a string.

Reply:
if refresh_token is str, then can we rename the one in sdk/identity/azure-identity/azure/identity/_credentials/authorization_code.py?

Reply:
Sure, but I prefer to punt it to a future PR because I intend to make more radical changes to `AadClient` that will affect the way `AuthorizationCodeCredential` interacts with the cache. I expect if I rename that variable in this PR, I'll soon revert the rename.